Release Godot 4.1.1 stable and publish release notes (#676)
godotengine/godot-website: collections/_article/maintenance-release-godot-4-1-1.md

---
title: "Maintenance release: Godot 4.1.1"
excerpt: "Hot on the heels of its parent release, here's Godot 4.1.1 — the first maintenance release of the 4.1 branch, with more stability and workflow improvements."
categories: ["release"]
author: Yuri Sizov
image: /storage/blog/covers/maintenance-release-godot-4-1-1.webp
image_caption_title: Gunforged
image_caption_description: A game by Firebelley
date: 2023-07-17 17:00:00
---

> The release of Godot 4.1 has been [just a couple of weeks ago](/article/godot-4-1-is-here), and we are ready for the first maintenance release, — addressing some of the bugs and annoyances that have slipped into the stable build.

Reviewer: ```suggestion
The release of Godot 4.1 was [just a couple of weeks ago](/article/godot-4-1-is-here) and we are ready for the first maintenance release — addressing some of the bugs and annoyances that have slipped into the stable build.
```

Some of the most important changes in this release include:

- Fix for a crash due to an infinite loop in `AnimationStateMachine` ([GH-79141](https://github.com/godotengine/godot/pull/79141)). It was a gnarly issue because it was easy to trigger with a bare minimum configuration. Now circular dependencies are correctly detected preventing infinite looping.

- Command-line export of C#/.NET projects should no longer drop random files ([GH-79173](https://github.com/godotengine/godot/pull/79173)). Your exports may have had arbitrary resources missing (not C# scripts, but images, for instance), if you were exporting your project with CLI. This should no longer happen.

- Custom export options which you can define with an `EditorExportPlugin` are now correctly restored on the editor restart ([GH-79025](https://github.com/godotengine/godot/pull/79025)). Previously the usability of this freshly added feature was limited due to data loss between sessions.

> - For Linux users there is a potential fix for freezes when interacting with menus on Wayland ([GH-79143](https://github.com/godotengine/godot/pull/79143)). This had been a hard to identify and debug issue, but our local Wayland enthusiasts managed to pinpoint the likely cause and validate that the unwanted behavior was addressed.

Reviewer: ```suggestion
- For Linux users there is a potential fix for freezes when interacting with menus on Wayland ([GH-79143](https://github.com/godotengine/godot/pull/79143)). This was a hard to identify and debug issue, but our local Wayland enthusiasts managed to pinpoint the likely cause and validate that the unwanted behavior was addressed.
```
